programmers-coding-test/kimjiwon/greedy/숫자카드게임.js

```diff
@@ -0,0 +1,23 @@
+function settingCard(row, column) {
+  const cards = [];
+  const compareMaxcards = [];
+  while (cards.length != column) {
+    cards.push(rowcards(row));
+  }
+
+  for (let i in cards) {
+    compareMaxcards.push(Math.min.apply(null, cards[i]));
+  }
+  return Math.max.apply(null, compareMaxcards);
+}
+
+console.log(settingCard(3, 4));
+
+//랜덤 카드 뽑기
+function rowcards(row) {
+  const numbers = new Set();
+  while (numbers.size != row) {
+    numbers.add(Math.floor(Math.random() * 9 + 1));
+  }
+  return [...numbers];
+}
```

Comment:
set 함수는 무언가를 반환하지 않는다.
세팅하는 부분과 반환하는 부분은 분리하는 것이 좋다.

```suggestion
const cards = [];

function setCards(row, column) {
  while (cards.length !== column) {
    cards.push(rowcards(row));
  }
}

function getMaxCard() {
  const minCards = [];
  for (let i in cards) {
    minCards.push(Math.min.apply(null, cards[i]));
  }
  return Math.max.apply(null, minCards);
}

setCards(3, 4);
console.log(getMaxCard());

//랜덤 카드 뽑기
function rowcards(row) {
  const numbers = new Set();
  while (numbers.size != row) {
    numbers.add(Math.floor(Math.random() * 9 + 1));
  }
  return [...numbers];
}
```

Reply:
Iterable 객체에 map 메서드를 이용하면 한 줄로도 문제를 해결할 수 있다.
```javascript
const getMaxCard = (cards) => Math.max(...cards.map(card => card.sort((a, b) => a - b)[0]));
console.log(`한 줄 풀이 : ${getMaxCard(cards)}`);
```

Reply:
오호 이런 방법이..